### PR TITLE
feat: update register modal

### PR DESCRIPTION
**Pull request by xyinshen**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/register_model/model_source.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  EuiLink,
+  EuiFormRow,
+  EuiComboBox,
+} from '@elastic/eui';
+
+import { useController, useFormContext } from 'react-hook-form';
+import { useMonitoring } from '../monitoring/use_monitoring';
+
+export const ModelSource = () => {
+  const { allExternalConnectors } = useMonitoring();
```
Comment (wanglam) [inline public/components/register_model/model_source.tsx:20]: `useMonitoring` has many other logic, how about separate another react hook to fetch all external connectors?

File: public/components/register_model/model_source.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const CONNECTOR_OPTIONS = allExternalConnectors?.map((item) => {
```
Comment (wanglam) [inline public/components/register_model/model_source.tsx:21]: Prefer rename CONNECTOR_OPTIONS to `connectorOptions`

File: public/components/register_model/model_source.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    return Object.assign({}, { label: item.name, value: item.description });
```
Comment (wanglam) [inline public/components/register_model/model_source.tsx:22]: I think, the value should be `item.id`

File: public/components/register_model/pretrainedmodel_select.tsx
Comment (wanglam) [inline public/components/register_model/pretrainedmodel_select.tsx:1]: Prefer rename file to `pretrained_model_select`

File: public/components/register_model/register_model.tsx [hunk 4]
```diff
@@ -232,15 +246,14 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
           const { config } = preTrainedModel;
           form.setValue('modelFileFormat', 'TORCH_SCRIPT');
           if (config.name) {
-            form.setValue('name', `huggingface/${config.name}`);
+            form.setValue('name', config.name);
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:249]: Why we need to remove huggingface.

File: public/components/register_model/pretrainedmodel_select.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          description: models[name].description,
+          checked: undefined,
+        }))
+      );
+    });
+    return () => {
+      subscribe.unsubscribe();
+    };
+  }, []);
+  const ShowRest = useCallback(
```
Comment (wanglam) [inline public/components/register_model/pretrainedmodel_select.tsx:55]: This function could be removed.

File: public/components/register_model/pretrainedmodel_select.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    [ShowRest]
+  );
+  useEffect(() => {
+    const selectedOption = modelRepoSelection.find((option) => option.checked === 'on');
+    if (selectedOption?.label) {
+      history.push(
+        `${generatePath(routerPaths.registerModel, { id: undefined })}/?type=import&name=${
+          selectedOption?.label
+        }&version=${selectedOption?.label}`
```
Comment (wanglam) [inline public/components/register_model/pretrainedmodel_select.tsx:78]: The version parameter is not exists. Should be removed.

File: public/components/register_model/pretrainedmodel_select.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+import { generatePath } from 'react-router-dom';
+import { modelRepositoryManager } from '../../utils/model_repository_manager';
+import { routerPaths } from '../../../common/router_paths';
+interface IItem {
+  label: string;
+  checked?: 'on' | undefined;
+  description: string;
+}
+interface Props {
+  getPreSelected: (val: boolean) => void;
```
Comment (wanglam) [inline public/components/register_model/pretrainedmodel_select.tsx:25]: Can be removed

File: public/components/register_model/pretrainedmodel_select.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    [getPreSelected]
+  );
+  const [modelRepoSelection, setModelRepoSelection] = useState<Array<EuiSelectableOption<IItem>>>(
+    []
+  );
+  const history = useHistory();
+  const onChange = useCallback(
+    (modelSelection: Array<EuiSelectableOption<IItem>>) => {
+      setModelRepoSelection(modelSelection);
+      ShowRest(true);
```
Comment (wanglam) [inline public/components/register_model/pretrainedmodel_select.tsx:68]: Can be removed.

File: public/components/register_model/register_model.tsx [hunk 9]
```diff
@@ -88,13 +89,21 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
     services: { chrome, notifications },
   } = useOpenSearchDashboards();
   const isLocked = useObservable(chrome?.getIsNavDrawerLocked$() ?? from([false]));
-
+  const [preSelected, setPreSelect] = useState(false);
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:92]: Can be removed

File: public/components/register_model/register_model.tsx [hunk 10]
```diff
@@ -88,13 +89,21 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
...
+  const getPreSelected = (val: boolean) => {
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:93]: Can be removed.

File: public/components/register_model/register_model.tsx [hunk 11]
```diff
@@ -326,14 +366,19 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
...
+                  <FormPartial />
+                )}
                 {FormPartial === ModelOverviewTitle || FormPartial === FileAndVersionTitle ? (
                   <EuiSpacer size="s" />
                 ) : (
                   <EuiSpacer size="xl" />
                 )}
               </React.Fragment>
             ))}
+            {formType === 'import' ? preSelected && formFooter : formFooter}
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:381]: Prefer change to
```tsx
{formType !== 'import' && !nameParams && formFooter}
```

File: public/components/register_model/register_model.tsx [hunk 12]
```diff
@@ -290,17 +314,33 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
...
-        <EuiLoadingSpinner aria-label="Model Form Loading" size="l" />
-      </EuiPanel>
-    );
-  }
-
+  const [checked, setChecked] = useState(false);
+  const onChange = (e: any) => {
+    setChecked(e.target.checked);
+  };
+  const formFooter = (
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:321]: I think these two checkboxes can be separate to two form partials. We can add them to the form partials array.

File: public/components/register_model/register_model.tsx [hunk 13]
```diff
@@ -88,13 +89,21 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
...
   } = useOpenSearchDashboards();
   const isLocked = useObservable(chrome?.getIsNavDrawerLocked$() ?? from([false]));
-
+  const [preSelected, setPreSelect] = useState(false);
+  const getPreSelected = (val: boolean) => {
+    setPreSelect(val);
+  };
   const formType = isValidModelRegisterFormType(typeParams) ? typeParams : 'upload';
-  const [preTrainedModelLoading, setPreTrainedModelLoading] = useState(formType === 'import');
   const partials =
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:97]: Prefer refactor to like these:
```
const partials = (() => {
    if (formType === 'import') {
      if (!nameParams) {
        return [PreTrainedModelSelect];
      }
      return [PreTrainedModelSelect, ModelDetailsPanel, ModelTagsPanel, ModelVersionNotesPanel];
    }
    if (formType === 'external') {
      return [
        ...(registerToModelId ? [] : [ModelDetailsPanel]),
        ...(registerToModelId ? [] : [ModelTagsPanel]),
        ModelSource,
        ModelVersionNotesPanel,
      ];
    }
    return [
      ...(registerToModelId ? [] : [ModelOverviewTitle]),
      ...(registerToModelId ? [] : [ModelDetailsPanel]),
      ...(registerToModelId ? [] : [ModelTagsPanel]),
      ...(registerToModelId ? [] : [FileAndVersionTitle]),
      ArtifactPanel,
      ConfigurationPanel,
      ...(registerToModelId ? [ModelTagsPanel] : []),
      ModelVersionNotesPanel,
    ];
  })();
```

File: public/components/register_model/register_model.tsx (hunk 4 above)
Comment (xyinshen) [reply public/components/register_model/register_model.tsx:249]: image
In the new figma , there is no huggingface.

File: public/components/register_model/pretrainedmodel_select.tsx (hunk 5 above)
Comment (xyinshen) [reply public/components/register_model/pretrainedmodel_select.tsx:55]: I design this function to control the rest showing after the pretrained model is seletcted. If removed, maybe this function will be failed.

File: public/components/register_model/register_model.tsx (hunk 4 above)
Comment (wanglam) [reply public/components/register_model/register_model.tsx:249]: As I remember, the backend need the `huggingface` prefix to register the pre-trained model. @zane-neo

**Pull request closed without merging**